op-service/eth/types.go

```diff
@@ -297,12 +290,16 @@ func (envelope *ExecutionPayloadEnvelope) CheckBlockHash() (actual common.Hash,
 		MixDigest:        common.Hash(payload.PrevRandao),
 		Nonce:            types.BlockNonce{}, // zeroed, proof-of-work legacy
 		BaseFee:          (*uint256.Int)(&payload.BaseFeePerGas).ToBig(),
+		WithdrawalsHash:  nil, // set below
+		BlobGasUsed:      (*uint64)(payload.BlobGasUsed),
+		ExcessBlobGas:    (*uint64)(payload.ExcessBlobGas),
 		ParentBeaconRoot: envelope.ParentBeaconBlockRoot,
+		RequestsHash:     envelope.RequestsHash,
 	}
 
-	if payload.IsthmusBlock() {
+	if payload.WithdrawalsRoot != nil {
 		header.WithdrawalsHash = payload.WithdrawalsRoot
```

Comment:
Why don't we just set `header.RequestsHash = types.EmptyRequestsHash` here in the Isthmus case, and not add the `RequestsHash` to the `ExecutionPayloadEnvelope`?